crates/ra_hir/src/nameres.rs

```diff
@@ -153,28 +154,102 @@ impl ModuleScope {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Default)]
+type ItemOrMacro = Either<PerNs<ModuleDef>, MacroDef>;
+
+trait ItemOrMacroExt {
```

Comment:
Perhaps it would be easier to just methods like `Either::left` and such, instead of adding a custom extension trait?

Reply:
Sure, Fixed